File: cli/src/cli/node.ts
import { NodeEntity } from '@protontech/drive-sdk';

export function findName(nodes: NodeEntity[], uid: string): string {
    for (const node of nodes) {
        if (node.uid === uid) {
            return getName(node);
        }
    }
    return uid;
}

export function getName(node: NodeEntity): string {
    const name = node.name.ok ? node.name.value : '';
    return name.length > 0 ? name : node.uid;
}

export function getClaimedSize(node: NodeEntity): number | undefined {
    const activeRevision = node.activeRevision?.ok ? node.activeRevision.value : undefined;
    return activeRevision?.claimedSize;
}

File: client/cs/src/Proton.Drive.Sdk/Nodes/Node.cs
using System.Text.Json.Serialization;
using Proton.Drive.Sdk.Events;
using Proton.Sdk;

namespace Proton.Drive.Sdk.Nodes;

public abstract record Node
{
    public required NodeUid Uid { get; init; }

    public required NodeUid? ParentUid { get; init; }

    [JsonIgnore]
    public DriveEventScopeId TreeEventScopeId => new(Uid.VolumeId);

    public required Result<string, ProtonDriveError> Name { get; init; }

    public required DateTime CreationTime { get; init; }

    public DateTime? TrashTime { get; init; }

    public required Result<Author, SignatureVerificationError> NameAuthor { get; init; }

    public required Result<Author, SignatureVerificationError> KeyAuthor { get; init; }

    public required OwnedBy OwnedBy { get; init; }

    public required bool IsShared { get; init; }

    public required bool IsSharedPublicly { get; init; }

    public required IReadOnlyList<ProtonDriveError> Errors { get; init; }
}
